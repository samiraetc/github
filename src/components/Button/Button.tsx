export const COMPONENT_ID = 'button';

type ButtonProps = {
  name: string;
  onClick?: () => void;
  className?: string;
};
const Button = ({ name, onClick, className = '' }: ButtonProps) => {
  return (
    <div className={`${COMPONENT_ID} ${className}`} id={`${COMPONENT_ID}`}>
      <button className={`${COMPONENT_ID}__action`} onClick={onClick}>
        {name}
      </button>
    </div>
  );
};

export default Button;
