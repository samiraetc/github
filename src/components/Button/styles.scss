.button {
  &__action {
    padding: 18px;
    color: $github-gray;
    background-color: $github-secondary-color;
    border: none;
    border-radius: $github-border-radius-20;
    width: 200px;
    box-shadow: 0 9px 18px rgba(0, 0, 0, 0.2);
    font-weight: 700;
    letter-spacing: 1px;
    text-decoration: none;
  }
}
