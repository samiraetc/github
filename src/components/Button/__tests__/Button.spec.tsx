import React from 'react';
import { configure, render } from '@testing-library/react';

import Button, { COMPONENT_ID } from '../Button';

configure({
  testIdAttribute: 'id',
});

describe('Button', () => {
  const onClick = jest.fn();
  test('should render Button', () => {
    const { asFragment, getByText, getByTestId } = render(<Button name="search" onClick={onClick} />);

    expect(getByText('search'));
    expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
