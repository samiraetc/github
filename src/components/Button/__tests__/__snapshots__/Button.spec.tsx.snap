// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<DocumentFragment>
  <div
    class="button "
    id="button"
  >
    <button
      class="button__action"
    >
      search
    </button>
  </div>
</DocumentFragment>
`;
